### ISSUE TITLE
MDX: React Components with Blank Lines

### ISSUE THREAD

**Issue by karlhorky** [prettier/prettier-vscode#1547]
### Summary

An extra blank line will be added after opening and closing tags of React components, if there are extra lines within the component.

This is a regression from a previous version.

![Kapture 2020-09-09 at 10 36 53](https://user-images.githubusercontent.com/1935696/92575269-69a33880-f288-11ea-8a9a-ea15645dd7c6.gif)

**Before**

```md
# Heading

<Component>

Markdown

</Component>

```

**After**

```diff
# Heading

<Component>

+
Markdown

</Component>

+
```

### Github Repository to Reproduce Issue

https://gist.github.com/karlhorky/f0a4a1d9d7b4be84ab0e4f5739fda71e

### Steps To Reproduce:

1.  This is step 1.
1.  This is step 2. All steps should start with '1.'

### Expected result

No extra lines

### Actual result

Extra lines after opening and closing tags

### Additional information

Potentially related to: https://github.com/prettier/prettier/issues/7041

Playground (with unexpected token issue mentioned in https://github.com/prettier/prettier/issues/7041): [Playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBiABACTgQwCYCWUA5gDpTkA8AwhALYAO0CMAfOeQLI4BOA1nggB3CqMoB6Wo2ax2UEABoQEBjALQAzslC8ewgAq8EWlDgA2QnAE8tSgEY8cYPnBgBlBk6LFkMHgFc4JTg6Ozg8PHCAGRwSfxxiOAAxCB46HBg1EmQQHH8YCEUQAAsYOjMAdWKCeA1PMDg3YxqCADcaqxywDVsQIg04Hhh9R2J05AAzcwGlACsNAA8AIUdnVzccOjgoojhJ6aCQeYW3bzM4AEV-CHh9sxmQTx4Bnhy6PAWihh4iGAqCPAwYrIAAcAAYlN8IAMKo4GDlvnAXq09koAI7XeAjFQmXIaAC0UDg4XCRR4cAxBHJIwS4yQU3uhwGdAIvgCTLOl0xe3pByUMBwdn+gOBSAATPzHAQzN4pHSQEiAKxFfwDAAqgpMDIerUCAEkoJFYG4wD9VABBQ1uGBWc53AYAXwdQA)

**VS Code Version**: 1.48.2

**Prettier Extension Version**: 5.5.0

**OS and version**: macOS Catalina 10.15.6

## Prettier Log Output

```
["INFO" - 10:38:44 AM] Formatting /Users/k/p/project/index.mdx
["INFO" - 10:38:44 AM] Using ignore file (if present) at /Users/k/p/course-material/.prettierignore
["INFO" - 10:38:44 AM] Using bundled version of prettier.
["INFO" - 10:38:44 AM] File Info:
{
  "ignored": false,
  "inferredParser": "mdx"
}
["INFO" - 10:38:44 AM] Detected local configuration (i.e. .prettierrc or .editorconfig), VS Code configuration will not be used
["INFO" - 10:38:44 AM] Prettier Options:
{
  "filepath": "/Users/k/p/project/index.mdx",
  "parser": "mdx",
  "proseWrap": "never",
  "singleQuote": true,
  "trailingComma": "all"
}
["INFO" - 10:38:44 AM] Formatting completed in 82.884081ms
```

**Comment by ntotten** [comment]
I am unable to reproduce this. I'll need more details like a full sample project with config, etc.

**Comment by karlhorky** [comment]
Here's a repo: https://github.com/karlhorky/vscode-prettier-mdx-bug-reproduction

I took all the configuration out and I am able to reproduce on both VS Code and VS Code Insider's Edition.

This is with v5.7.1 of the extension (using the built-in version of Prettier).

**Comment by ntotten** [comment]
Ah yeah, this is a bug with prettier, not VS Code. I am able to get the same result through the command line `prettier index.mdx --write` on version 2.1.2 of prettier. Please open a new bug on the main repo: https://github.com/prettier/prettier

**Comment by karlhorky** [comment]
Oh, sorry about that!

I thought that there was special handling in `prettier-vscode` for this, since the playground errors out on that code.

Looks like there's already an open issue: https://github.com/prettier/prettier/issues/9260

**Comment by karlhorky** [comment]
Can confirm that this is fixed in Prettier `2.2.0`. You may need to upgrade your local Prettier version in your project if it's not working.
